src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -38,6 +38,51 @@ private static void VerifyIndexCall(IMethodSymbol symbol, string methodName, str
             Assert.Equal(containingTypeName, symbol.ContainingType.Name);
         }
 
+        [Fact]
+        public void PatternIndexArray()
+        {
+            var src = @"
+class C
+{
+    static int M1(int[] arr) => arr[^1];
+    static char M2(string s) => s[^1];
+}
+";
+            var verifier = CompileAndVerifyWithIndexAndRange(src);
+            // Code gen for the following two should look basically
+            // the same, except that string will use Length/indexer
+            // and array will use ldlen/ldelem, and string may have
+            // more temporaries
+            verifier.VerifyIL("C.M1", @"
+{
+  // Code size        8 (0x8)
+  .maxstack  3
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  ldlen
+  IL_0003:  conv.i4
+  IL_0004:  ldc.i4.1
+  IL_0005:  sub
+  IL_0006:  ldelem.i4
+  IL_0007:  ret
+}");
+            verifier.VerifyIL("C.M2", @"
+{
+  // Code size       17 (0x11)
+  .maxstack  3
+  .locals init (int V_0)
```

Comment:
any reason we capture this local?

Reply:
It looks like it's only used for a conversion below in the `stloc/ldloc` pair. Probably unnecessary, but the JIT gets rid of that, so it's no different in performance. Not sure why codegen makes the local but it may be possible to improve.